src/tensorinfo.cpp

```diff
@@ -30,42 +30,42 @@ namespace ovms {
 TensorInfo::TensorInfo(const std::string& name,
     const Precision& precision,
     const Shape& shape) :
-    name(name),
-    mapping(""),
-    precision(precision),
-    shape(shape),
-    layout(Layout::getDefaultLayout()) {}
+    TensorInfo(name, "", precision, shape, Layout::getDefaultLayout()) {}
 
 TensorInfo::TensorInfo(const std::string& name,
     const Precision& precision,
     const shape_t& shape) :
-    name(name),
-    mapping(""),
-    precision(precision),
-    shape(shape),
-    layout(Layout::getDefaultLayout()) {
-}
+    TensorInfo(name, "", precision, shape, Layout::getDefaultLayout()) {}
 
 TensorInfo::TensorInfo(const std::string& name,
     const ovms::Precision& precision,
     const shape_t& shape,
     const Layout& layout) :
-    name(name),
-    mapping(""),
-    precision(precision),
-    shape(shape),
-    layout(layout) {
-}
+    TensorInfo(name, "", precision, shape, layout) {}
+
+TensorInfo::TensorInfo(const std::string& name,
+    const std::string& mapping,
+    const Precision& precision,
+    const shape_t& shape) :
+    TensorInfo(name, mapping, precision, shape, Layout::getDefaultLayout()) {}
 
 TensorInfo::TensorInfo(const std::string& name,
     const ovms::Precision& precision,
     const Shape& shape,
     const Layout& layout) :
-    name(name),
-    mapping(""),
-    precision(precision),
-    shape(shape),
-    layout(layout) {
+    TensorInfo(name, "", precision, shape, layout) {}
+
+TensorInfo::ProcessingHint TensorInfo::getProcessingHint() const {
+    size_t expectedDimsForImage = this->influencedByDemultiplexer ? 5 : 4;
+    if (this->shape.size() == 2 && this->precision == ovms::Precision::U8 && !this->influencedByDemultiplexer) {
+        return TensorInfo::ProcessingHint::STRING_2D_U8;
+    } else if (this->shape.size() == 1 && this->precision == ovms::Precision::U8 && !this->influencedByDemultiplexer) {
+        return TensorInfo::ProcessingHint::STRING_1D_U8;
+    } else if (this->shape.size() == expectedDimsForImage) {
+        return TensorInfo::ProcessingHint::IMAGE;
+    } else {
+        return TensorInfo::ProcessingHint::NO_PROCESSING;
+    }
```

Comment:
1) This could be checked once during construction not once per request
2) Do we expect that this may be configurable at some time in future? It looks to me that this is part of model configuration, and we only should store here information about what the preprocessing hint would be.
At some point we may need to incorporate Model API pre/post processing into OVMS

Reply:
1. Agree
2. This hint is used for validation purposes as well, so overwriting it by some other entity would break the validation logic. Example: We validate incoming string batchsize for 2d endpoints because we **can** deduce batch from endpoints first dimension. Overwriting hint to 1d would miss the validation calls and also crash during conversion since we do not perform the checks at each stage to save time.

Reply:
Moved to constructor. Also made entire TensorInfo const and removed setters.